sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -620,8 +620,17 @@ object SparkConnectClient {
      * Configure the builder using the env SPARK_REMOTE environment variable.
      */
     def loadFromEnvironment(): Builder = {
+      lazy val isAPIModeConnect = Option(System.getProperty("spark.api.mode"))
+        .exists(_.toLowerCase(Locale.ROOT) == "connect")
       Option(System.getProperty("spark.remote")) // Set from Spark Submit
         .orElse(sys.env.get(SparkConnectClient.SPARK_REMOTE))
+        .orElse {
+          if (isAPIModeConnect) {
+            Option(System.getProperty("spark.master")).orElse(sys.env.get("MASTER"))
```

Comment:
A spark master URL is not compatible with a connect URL... How is this supposed to work. More generally, why should the client have to deal with spark master URLs, at this point in time you should be able to pass a valid remote URL.

Reply:
Oh yeah, this can be removed.

Reply:
Actually for now, we need this because we directly start ConnectRepl for Spark Connect shell. Once we have a layer for Spark Connect (e.g., at https://github.com/apache/spark/pull/49053), we would be able to separate it out.